src/types.rs

```diff
@@ -1349,9 +1349,20 @@ pub struct GraphDbRequest {
 
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub enum DefineResourceType {
-    Namespace { name: String },
-    Database { name: String },
-    Table { name: String },
+    Namespace {
+        name: String,
+    },
+    Database {
+        name: String,
+    },
+    Table {
+        name: String,
+    },
+    Index {
+        name: String,
+        table: String,
+        columns: Vec<String>,
```

Comment:
we should add the analyzer as a param here. ascii may not be the only one and should be a param.

Reply:
https://docs.surrealdb.com/docs/surrealql/statements/define/indexes#statement-syntax

we should support each of these:
```surql
DEFINE INDEX @name ON [ TABLE ] @table [ FIELDS | COLUMNS ] @fields
    [ UNIQUE | SEARCH ANALYZER @analyzer [ BM25 [(@k1, @b)] ] [ HIGHLIGHTS ] ]
```